### PR TITLE
ocamlPackages: init diet, irmin-chunk, mirage-block-combinators, mirage-block-ramdisk, mirage-block-unix, wodan

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change

I wanted to play around with the `wodanc` utility and hit some of the dependencies.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/diet/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  minimumOCamlVersion = "4.03";
+
+  propagatedBuildInputs = [ stdlib-shims ];
+
+  doCheck = true;
+  checkInputs = [ ounit ];
+
+  meta = with lib;
+    src.meta // {
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/diet/default.nix:21]: This `src.meta` looks wrong… For instance, there is a surprising `name` attribute.

File: pkgs/development/ocaml-modules/diet/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      description = "Simple implementation of Discrete Interval Encoding Trees";
+      license = licenses.isc;
+      maintainers = [ lib.maintainers.ehmry ];
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/diet/default.nix:24]: `lib` is redundant here.

**Review by vbgl — CHANGES_REQUESTED** [review]

File: pkgs/development/ocaml-modules/irmin/chunk.nix [hunk 3]
```diff
@@ -0,0 +1,18 @@
+{ lib, buildDunePackage, io-page-unix, irmin, irmin-mem, irmin-test }:
+
+buildDunePackage rec {
+
+  pname = "irmin-chunk";
+  inherit (irmin) version src useDune2;
+
+  propagatedBuildInputs = [ io-page-unix irmin ];
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/irmin/chunk.nix:8]: Is `io-page-unix` really needed?

File: pkgs/development/ocaml-modules/wodan/default.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    lru
+    io-page
+    mirage-block
+  ];
+
+  meta = with lib; {
+    inherit (src.meta) homepage;
+    description = "A flash-friendly, safe and flexible filesystem library";
+    license = licenses.isc;
+    maintainers = [ lib.maintainers.ehmry ];
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/wodan/default.nix:38]: `lib` is redundant here.

File: pkgs/development/ocaml-modules/wodan/irmin.nix [hunk 5]
```diff
@@ -0,0 +1,19 @@
+{ lib, buildDunePackage, irmin-chunk, irmin-git, irmin-unix
+, mirage-block-ramdisk, mirage-block-unix, wodan }:
+
+buildDunePackage rec {
+  pname = "wodan-irmin";
+  inherit (wodan) version src useDune2;
+
+  propagatedBuildInputs = [
+    irmin-chunk
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/wodan/irmin.nix:9]: `io-page-unix` might be missing here.

File: pkgs/development/ocaml-modules/irmin/chunk.nix (hunk 3 above)
Comment (ehmry) [reply pkgs/development/ocaml-modules/irmin/chunk.nix]: No? I can build it without, I'm not sure how it got in there.

File: pkgs/development/ocaml-modules/wodan/irmin.nix (hunk 5 above)
Comment (ehmry) [reply pkgs/development/ocaml-modules/wodan/irmin.nix:9]: Builds anyway?

Comment (vbgl) [reply pkgs/development/ocaml-modules/wodan/irmin.nix:9]: @GrahamcOfBorg build ocamlPackages.wodan-irmin

**Comment by vbgl** [comment]
Merged `diet` into `master` as 3d7d7b8af3e3ba8c0c009e813b6b06f376b14349.

**Comment by vbgl** [comment]
All commits but the last one have been merged (`irmin-chunk` is added by 9b2ddc9ad0b528513b4fa17ee4695a9055e0a15c).

Unfortunately, `wodan-irmin` won’t build:

> Error: Library "io-page-unix" not found.

File: pkgs/development/ocaml-modules/mirage-block-unix/default.nix [hunk 6]
```diff
@@ -15,7 +15,8 @@ buildDunePackage rec {
 
   minimumOCamlVersion = "4.06";
 
-  propagatedBuildInputs = [ cstruct-lwt logs mirage-block rresult uri ];
+  propagatedBuildInputs =
+    [ cstruct-lwt io-page-unix logs mirage-block rresult uri ];
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/mirage-block-unix/default.nix:19]: I don’t understand this change: the `mirage-block-unix` library does not use `io-page-unix` in any way, does it?

Comment (ehmry) [reply pkgs/development/ocaml-modules/mirage-block-unix/default.nix:19]: Its listed as a dependency, but somehow not used during the checkPhase: https://opam.ocaml.org/packages/mirage-block-unix/

Comment (vbgl) [reply pkgs/development/ocaml-modules/mirage-block-unix/default.nix:19]: Does this mean that there is a mistake in the page you’ve linked?

Comment (ehmry) [reply pkgs/development/ocaml-modules/mirage-block-unix/default.nix:19]: Well it is listed in the opam and dune metadata of mirage-block-unix.

**Comment by vbgl** [comment]
Merged into master as 93e31f2a78db66ecf3aa497c7fbed758655eea2e

**Pull request closed without merging**